### DISCUSSION TITLE
Optimization time interval

### DISCUSSION THREAD

**Discussion by woodjmichael** [NREL/REopt-API-Analysis#137] [8 - API: Application Programming Interface]
- [x] I searched the thread for similar questions
- [x] I am using the most recent version of REopt-Analysis-Scripts

How can I force the optimizer to work on a 15-minute interval, rather than down sampling to 1-hour interval? It seems to me my data is always being down sampled to 1-hour.

According to the [manual](https://reopt.nrel.gov/tool/reopt-user-manual.pdf) this is possible with the API if all input series are of length 30504 -- I confirm all of mine are.

However, when I POST and analyze results (making sure status is `optimal`) I get the identical financial results if I pass all 8760-length series to the POST, whereas we would expect at least some small differences in energy cost, etc. Also, the output series (battery dispatch etc) are all 8760-length regardless what length I pass to the POST (if the optimizer is working on 15-minute vectors we'd expect the output battery vectors to be 35040-length).

For simplicity's (speed) sake I'm forcing a single size solar and battery.

My POST:
```python
post = {
    "Site": {
        "longitude": 45.505890265478996, 
        "latitude": 9.160999024421262
    },
    "PV": {
        "production_factor_series":list(pd.read_csv('solar_profiles/jplv4_solarPU_aut19_35040.csv',
                                                    header=None).values.flatten()),
        "array_type": 0,
        "min_kw":48,
        "max_kw":48,
        "can_net_meter":False,
        "installed_cost_per_kw":0,
    },
    "ElectricLoad": {
        "loads_kw":list(pd.read_csv('load_profiles/jplv4_load_aut19_35040.csv',
                                    header=None).values.flatten())
    },
    "ElectricStorage": {
        "min_kw":48,
        "max_kw":48,
        "min_kwh":42,#140
        "max_kwh":42,#140
        "can_grid_charge":True,
        "installed_cost_per_kw":0,
        "installed_cost_per_kwh":0,
        "soc_min_fraction":0.0,
        "soc_max_fraction":1.0,
        "soc_init_fraction":0.0,
    },

    "ElectricTariff": {
        "tou_energy_rates_per_kwh":list(pd.read_csv('electric_rates/jplv4_price-buy_aut19_35040.csv',
                                    header=None).values.flatten()),
        "wholesale_rate":list(pd.read_csv('electric_rates/jplv4_price-sell_aut19_35040.csv',
                                    header=None).values.flatten())
    },
    "ElectricUtility":{
        "net_metering_limit_kw":0,
        
    },
    "Financial": {
        "elec_cost_escalation_rate_fraction": 0.026,
        "owner_discount_rate_fraction": 0.081,
        "analysis_years": 1,
        "offtaker_tax_rate_fraction": 0.4,
        "owner_tax_rate_fraction": 0.4,
        "om_cost_escalation_rate_fraction": 0.025
    }
}
```

**Comment by adfarth** [comment]
Hello! Can you try setting `Settings : { time_steps_per_hour = 4} ` in your post?

**Comment by woodjmichael** [comment]
I tried that originally but I always got:

api_response['status']=='error'

WIth error message:

{'errors': {'core_electric_tariff.jl_397': [' Export rates do not have
correct number of entries. Must be 35040 or 8760.']}

So I removed 'wholesale_rates' from ElectricTariff. Worked fine. I did
double check that my 'wholesale_rates' series was length 30504. I also
tried 'export_rate_beyond_net_metering_limit' and checked the length to be
35040, same error message. I do need that export price vector one way or
another (no net metering).

This json file uses the PGE_E20.json tariff. Same error message as above.
I've also tried various configurations of 'can_net_meter' and
'net_metering_limit_kw' just in case 'wholesale_rates' is somehow
incompatible with either.

Thanks for your help.
